File: fstp-core-bo/resource/com/purefun/fstp/example/bo/otw/ExampleBO_OTW.java
package com.purefun.fstp.example.bo.otw;

import com.purefun.fstp.example.bo.ExampleBO;
import com.purefun.fstp.example.bo.pro.ExampleBO_PRO;
import com.google.protobuf.InvalidProtocolBufferException;
import com.purefun.fstp.core.bo.commom.ICommom_OTW;
import com.google.protobuf.Any;

public class ExampleBO_OTW implements ICommom_OTW {
    ExampleBO_PRO.ExampleBO.Builder builder = null;
    ExampleBO bo = null;

    public ExampleBO_OTW() {
        builder = ExampleBO_PRO.ExampleBO.newBuilder();
        bo= new ExampleBO();
        builder.setUuid(bo.uuid);
        builder.setBoid(bo.boid);
        builder.setDestination(bo.destination);
    }

    public ExampleBO_OTW(byte[] message) throws InvalidProtocolBufferException {
        builder = ExampleBO_PRO.ExampleBO.newBuilder();
        bo= new ExampleBO();
        ExampleBO_PRO.ExampleBO receive = ExampleBO_PRO.ExampleBO.parseFrom(message);
        setName(receive.getName());
        setAge(receive.getAge());
        setUuid(receive.getUuid());
        setBoid(receive.getBoid());
        setDestination(receive.getDestination());
    }

    public ExampleBO_OTW(ExampleBO bofrom){
        builder = ExampleBO_PRO.ExampleBO.newBuilder();
        bo= new ExampleBO();
        setName(bofrom.name);
        setAge(bofrom.age);
        setUuid(bofrom.uuid);
        setBoid(bofrom.boid);
        setDestination(bofrom.destination);
    }

    public byte[] serial() {
        return builder.build().toByteArray();
    }

    @Override
    public com.google.protobuf.GeneratedMessageV3.Builder getBuilder() { 
        return builder;
    }

    @Override
    public ExampleBO getBo() { 
        return bo;
    }

    public java.lang.String getName() {
        return builder.getName();
    }

    public void setName(java.lang.String name) {
        bo.name = name;
        builder.setName(name);
    }

    public int getAge() {
        return builder.getAge();
    }

    public void setAge(int age) {
        bo.age = age;
        builder.setAge(age);
    }

    public java.lang.String getUuid() {
        return builder.getUuid();
    }

    public void setUuid(java.lang.String uuid) {
        bo.uuid = uuid;
        builder.setUuid(uuid);
    }

    public long getBoid() {
        return builder.getBoid();
    }

    public void setBoid(long boid) {
        bo.boid = boid;
        builder.setBoid(boid);
    }

    public java.lang.String getDestination() {
        return builder.getDestination();
    }

    public void setDestination(java.lang.String destination) {
        bo.destination = destination;
        builder.setDestination(destination);
    }

    public String toString() {
        return "ExampleBO_OTW ["+
            "uuid = " + getUuid() +"," +
            "boid = " + getBoid() +"," +
            "destination = " + getDestination() +"," +
            "name = " + getName() +"," +
            "age = " + getAge() +"," +
         "]";
    }
}

File: fstp-python/com/purefun/fstp/example/bo/otw/ExampleBO_OTW.py
from core.common.ICommon_OTW import ICommon_OTW
from com.purefun.fstp.example.bo.pro import ExampleBO_pb2
from com.purefun.fstp.example.bo.model.ExampleBO import ExampleBO

class ExampleBO_OTW(ICommon_OTW):

    def __init__(self, byteMsg = None):
        self._bo_pro = ExampleBO_pb2.ExampleBO()
        self._bo = ExampleBO()

        if byteMsg is not None:
            self._bo_pro.ParseFromString(byteMsg)
            self.__setDataFromPB()
        else:
            self.__setDataFromBO()

    def __setDataFromBO(self):
        self._bo_pro.name = self._bo.name
        self._bo_pro.age = self._bo.age
        self._bo_pro.uuid = self._bo.uuid
        self._bo_pro.boid = self._bo.boid
        self._bo_pro.destination = self._bo.destination

    def __setDataFromPB(self):
        self._bo.name = self._bo_pro.name
        self._bo.age = self._bo_pro.age
        self._bo.uuid = self._bo_pro.uuid
        self._bo.boid = self._bo_pro.boid
        self._bo.destination = self._bo_pro.destination

    def getBO(self):
        return self._bo

    def getProBO(self):
        return self._bo_pro

    def getName(self):
        return self._bo.name

    def setName(self, name):
        self._bo.name = name
        self._bo_pro.name = name

    def getAge(self):
        return self._bo.age

    def setAge(self, age):
        self._bo.age = age
        self._bo_pro.age = age

    def getUuid(self):
        return self._bo.uuid

    def setUuid(self, uuid):
        self._bo.uuid = uuid
        self._bo_pro.uuid = uuid

    def getBoid(self):
        return self._bo.boid

    def setBoid(self, boid):
        self._bo.boid = boid
        self._bo_pro.boid = boid

    def getDestination(self):
        return self._bo.destination

    def setDestination(self, destination):
        self._bo.destination = destination
        self._bo_pro.destination = destination

    def toString(self):
        return "ExampleBO_OTW ["+"name = " + self.getName() +"," +"age = " + str(self.getAge()) +"," +"uuid = " + self.getUuid() +"," +"boid = " + str(self.getBoid()) +"," +"destination = " + self.getDestination() +"," +"]"
